File: codegen/results/v002/opus/20260411_143810/typescript/001_csv_to_json/solution.ts
import fs from 'fs';
import path from 'path';

const filePath = path.resolve('input/data.csv');
const content = fs.readFileSync(filePath, 'utf-8');
const lines = content.split('\n').filter(line => line.trim() !== '');
const [, ...rows] = lines;

const result = rows.map(row => {
  const [Name, Age, Email, Score] = row.split(',').map(f => f.trim());
  return {
    Name,
    Age: parseInt(Age, 10),
    Email,
    Score: parseFloat(Score),
  };
});

process.stdout.write(JSON.stringify(result));

File: codegen/results/v002/opus/20260411_143810/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"log"
	"os"
	"strconv"
)

type Record struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	f, err := os.Open("input/data.csv")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	r := csv.NewReader(f)
	rows, err := r.ReadAll()
	if err != nil {
		log.Fatal(err)
	}

	var records []Record
	for _, row := range rows[1:] {
		age, err := strconv.Atoi(row[1])
		if err != nil {
			log.Fatal(err)
		}
		score, err := strconv.ParseFloat(row[3], 64)
		if err != nil {
			log.Fatal(err)
		}
		records = append(records, Record{
			Name:  row[0],
			Age:   age,
			Email: row[2],
			Score: score,
		})
	}

	if records == nil {
		records = []Record{}
	}

	enc := json.NewEncoder(os.Stdout)
	enc.SetIndent("", "  ")
	if err := enc.Encode(records); err != nil {
		log.Fatal(err)
	}
}